pilot/pkg/bootstrap/istio_ca.go

```diff
@@ -477,10 +477,13 @@ func (s *Server) createIstioRA(client kubelib.Client,
 	caCertFile := path.Join(ra.DefaultExtCACertDir, constants.CACertNamespaceConfigMapDataName)
 	certSignerDomain := opts.CertSignerDomain
 	_, err := os.Stat(caCertFile)
-	if err != nil && certSignerDomain == "" {
-		caCertFile = defaultCACertPath
-	} else {
-		caCertFile = ""
+	if err != nil {
```

Comment:
should check `os.IsNotExist(err)`. if it fails for other reasons its a real error.

Reply:
@howardjohn thanks for the comment - so if it fails for other reasons we should just return an error?
```
	if err != nil {
		if !os.IsNotExist(err) {
			return nil, fmt.Errorf("failed to get file info: %v", err)
		}
		
		if certSignerDomain == "" {
			log.Infof("CA cert file %q not found, using %q.", caCertFile, defaultCACertPath)
			caCertFile = defaultCACertPath
		} else {
			log.Infof("CA cert file %q not found.", caCertFile)
			caCertFile = ""
		}
	}
```